### PR TITLE
Primitive Historical Profiler Implementation

### PR DESCRIPTION
**Pull request by stevensecreti**
As the title says, a basic implementation of the HistoricalProfiler class.

Files added:
 - `historical_profiler.py:` Contains class implementation with basic CRUD methods as well as a method for `consecutive difference reports` and a `consecutive difference min max report` that gives the minimum and maximum differences between all profiles for each (numerical) key in the report dict. The latter report provides the basis to be used in conjunction with `great_expectations` to automatically produce threshold bounds when checking that a new dataset's profile has only drifted within expected margins overtime. 
 - `historical_profiler.ipynb:` Contains examples and brief explanations of each method in the `HistoricalProfiler` class
 - `ny_climate.csv:` contains data from Kaggle ([found here](https://www.kaggle.com/datasets/die9origephit/temperature-data-albany-new-york?select=monthly_data.csv)) of climate data in the state of NY over the time frame 2015-2022. Each year is separated into its own dataset in the example notebook, which are used to initialize a `HistoricalProfiler()` object.

**WIP** - Next step is to add tests

### CODE REVIEW

File: dataprofiler/profilers/__init__.py [hunk 1]
```diff
@@ -10,3 +10,4 @@
 from .profile_builder import Profiler, StructuredProfiler, UnstructuredProfiler
 from .text_column_profile import TextColumn
 from .unstructured_labeler_profile import UnstructuredLabelerProfile
+from .historical_profiler import HistoricalProfiler
```
Comment (taylorfturner) [inline dataprofiler/profilers/__init__.py:13]: new line EOF

File: dataprofiler/tests/data/csv/ny_climate.csv [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+2021,5.4,12,0.8,13,-11,-0.2,0.84,22-23,2006-09-08
+2021,6.3,12,0.3,-23,-31,-0.5,2.04,23-24,2006-09-08
+2021,4.9,15,7.1,-2,-222,5.6,1.65,04-04,2006-09-08
+2021,6.3,10,-0.9,0,26,0.6,1.17,12-12,2006-09-08
+2021,7.8,17,7.6,0,-238,7,0.67,25-26,2006-09-08
+2022,8.7,10,-2.9,0,83,-1.6,0.66,16-17,2006-09-08
+2022,10.3,10,1.6,0,-46,3,1.25,03-04,2006-09-08
+2022,9.2,14,2.3,0,-72,2.8,1.01,31-31,2006-09-08
+2022,9.6,14,0,-4,-5,0.1,2.24,07-08,2006-09-08
+2022,7.2,10,5,58,-98,5.7,0.52,16-17,2006-09-08
```
Comment (taylorfturner) [inline dataprofiler/tests/data/csv/ny_climate.csv:88]: new line EOF

File: dataprofiler/__init__.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
     UnstructuredProfiler,
 )
 from .profilers.profiler_options import ProfilerOptions
+from .profilers.historical_profiler import HistoricalProfiler
```
Comment (taylorfturner) [inline dataprofiler/__init__.py:17]: did you run the `pre-commit` on this? I think `isort` would put this on line 11

File: dataprofiler/profilers/historical_profiler.py [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+from .utils import(
+    find_diff_of_dates,
+    find_diff_of_dicts_with_diff_keys,
+    find_diff_of_lists_and_sets,
+    find_diff_of_matrices,
+    find_diff_of_numbers,
+    find_diff_of_strings_and_bools,
+)
+
+class HistoricalProfiler():
```
Comment (taylorfturner) [inline dataprofiler/profilers/historical_profiler.py:18]: haven't reviewed this but be sure to run `pre-commit` as documented [here](https://github.com/capitalone/DataProfiler#pre-commit)

**Review by JGSweets — COMMENTED** [review]
For the Notebook, let's clear the cell outputs before uploading

File: dataprofiler/profilers/historical_profiler.py [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+
+        This method will also remove the last value stored in the lists in list_dict
+        in the case that adding the new value will result in the list becoming
+        larger than `max_length` set in options.
+        """
+        for key, val in list_dict.items():
+            if key in self.options["exclude_keys"]:
+                continue
+            if key not in values_dict:
+                print(
```
Comment (JGSweets) [inline dataprofiler/profilers/historical_profiler.py:252]: use logger like in profile_builder

Comment (stevensecreti) [reply dataprofiler/profilers/historical_profiler.py:252]: Realized this would be better accomplished as a warning

**Comment by stevensecreti** [comment]
Added basic tests in commit 107b74e

**Review by taylorfturner — APPROVED** [review]

**Review by taylorfturner — APPROVED** [review]

**Review by JGSweets — APPROVED** [review]

**Pull request merged**